static/js/app/cloudfoundry.js

```diff
@@ -0,0 +1,57 @@
+(function() {
+    // CloudFoundry Service
+    var cloudfoundry = function($http) {
+
+        // Get current authentication status from server
+        var getAuthStatus = function() {
+            return $http.get('/v2/authstatus')
+                .then(function(response) {
```

Comment:
Yep the first is for a success and the second is for a failure. In this case we need two of the same functions because we want the site to return the status in both cases. Still, I cleared up the code a bit.